aeron-driver/src/main/c/aeron_driver_conductor.c

```diff
@@ -3225,6 +3230,48 @@ int aeron_driver_conductor_link_subscribable(
     return result;
 }
 
+void aeron_driver_subscribable_state(
+    aeron_subscribable_t *subscribable,
+    aeron_tetherable_position_t *tetherable_position,
+    aeron_subscription_tether_state_t state,
+    int64_t now_ns)
+{
+    if (tetherable_position->state != state)
+    {
+        if (AERON_SUBSCRIPTION_TETHER_RESTING == state)
+        {
+            subscribable->num_resting++;
+        }
+        else if (AERON_SUBSCRIPTION_TETHER_RESTING == tetherable_position->state)
+        {
+            subscribable->num_resting--;
+        }
+    }
+
+    tetherable_position->state = state;
+    tetherable_position->time_of_last_update_ns = now_ns;
+}
+
+size_t aeron_driver_subscribable_working_position_count(aeron_subscribable_t *subscribable)
+{
+    size_t count = 0;
+
+    for (int lastIndex = subscribable->length - 1, i = lastIndex; i >= 0; i--)
+    {
+        if (AERON_SUBSCRIPTION_TETHER_RESTING != subscribable->array[i].state)
+        {
+            count++;
+        }
+    }
+
```

Comment:
Is this just `subscribable->length - subscribable->num_resting`?

Reply:
Yes, missed that optimisation.